modules/likes.php

```diff
@@ -557,24 +557,85 @@ function admin_bar_likes() {
 	}
 }
 
+/**
+ * Callback to get the value for the jetpack_likes_enabled field.
+ */
+function jetpack_post_likes_get_value( array $post ) {
+	$post_likes_switched = get_post_meta( $post['id'], 'switch_like_status', true );
+
+	/** This filter is documented in modules/likes.php */
+	$sitewide_likes_enabled = (bool) apply_filters( 'wpl_is_enabled_sitewide', ! get_option( 'disabled_likes' ) );
+
+	/** This filter is documented in class.json-api-endpoints.php */
+	$is_jetpack = true === apply_filters( 'is_jetpack_site', false, get_current_blog_id() );
+
+	if ( $is_jetpack ) {
+		// an empty string: post meta was not set, so go with the global setting
+		if ( "" === $post_likes_switched ) {
+			return $sitewide_likes_enabled;
+		}
+
+		// user overrode the global setting to disable likes
+		elseif ( "0" === $post_likes_switched ) {
+			return false;
+		}
+
+		// user overrode the global setting to enable likes
+		elseif ( "1" === $post_likes_switched ) {
+			return true;
+		}
+
+		// no default fallback, let's stay explicit
+	}
+
+	// WPCOM sites always set the meta to 1 if the user is flipping the global setting
+	return $post_likes_switched xor $sitewide_likes_enabled;
+}
+
+/**
+ * Callback to set switch_like_status post_meta when jetpack_likes_enabled is updated.
+ */
+function jetpack_post_likes_update_value( $enable_post_likes, $post_object ) {
+	/** This filter is documented in modules/likes.php */
+	$sitewide_likes_enabled = (bool) apply_filters( 'wpl_is_enabled_sitewide', ! get_option( 'disabled_likes' ) );
+
+	$should_switch_status = $enable_post_likes xor $sitewide_likes_enabled;
+
+	/** This filter is documented in class.json-api-endpoints.php */
+	$is_jetpack = true === apply_filters( 'is_jetpack_site', false, get_current_blog_id() );
+
+	if ( $should_switch_status ) {
+		// WPCOM sites set the meta to 1 if the user is flipping the global setting
+		$meta_value = 1;
+		if ( $is_jetpack ) {
+			// set the meta to 0 if the user wants to disable likes, 1 if user wants to enable
+			$meta_value = $enable_post_likes;
+		}
```

Comment:
this isn't a blocker, but in this pattern…

```php
$var = A;
if ( $condition ) {
	$var = B;
}
```

…I find it help to assign with one non-mutating binding and then we don't have to track reassignments in our head when reading the code, plus it's shorter and there's less visual noise.

```php
// Jetpack is a force-override while on WPCOM we indicate that we should flip
$meta_value = $is_jetpack ? $enable_post_likes : 1;
```